python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py

```diff
@@ -286,9 +292,8 @@ def test_materialize_no_selection(
     )
     assert not result.success  # fail_tests_model fails
     assert len(result.get_asset_materialization_events()) == 10
-    assert len(result.get_asset_check_evaluations()) == 24
-    # one singular test
-    assert len(result.get_asset_observation_events()) == 1
+    assert len(result.get_asset_check_evaluations()) == 26
+    assert len(result.get_asset_observation_events()) == 2
```

Comment:
To explain the math here for asset check evaluations:

- Two asset checks added (one singular test with a single dependency, and one singular test with a configured asset key)

And for asset observations:

- One asset observation got removed since a singular test is now an asset check.
- Two asset observations got added for the singular test that does not have a configured asset key. It references two models, so two observations are emitted.